rl_bakery/agent_trainer/agents.py

```diff
@@ -0,0 +1,173 @@
+import abc
+import tensorflow as tf
+from tf_agents.utils import common
+from tf_agents.networks import q_network
+from tf_agents.agents.dqn import dqn_agent
+from tf_agents.agents import ddpg
+from tf_agents.trajectories import time_step as ts
+import logging
+
+from rl_bakery.agent_trainer.config import Optimizer
+
+logger = logging.getLogger(__name__)
+
+
+class AgentTrainer(metaclass=abc.ABCMeta):
+    """
+    This set ups an Agent to be trained.
+    """
+
+    def __init__(self, data_spec, config):
+        self._data_spec = data_spec
+        self._config = config
+
+    @abc.abstractmethod
+    def init_agent(self):
+        pass
+
+
+class TFAgentTrainer(AgentTrainer):
+
+    def __init__(self, data_spec, config):
+        super().__init__(data_spec, config)
+
+    @abc.abstractmethod
+    def _init_tfagent(self, optimizer):
+        pass
+
+    def _get_optimizer(self, optimizer_config):
+
+        # TODO use tf2 optimizers, support other optimizers
+        if optimizer_config.optimizer == Optimizer.Adam:
+            return tf.compat.v1.train.AdamOptimizer(learning_rate=optimizer_config.learning_rate)
```

Comment:
These optimizers have man other parameters that can be set which are not exposed in this setup such as _beta1_, _beta2_.

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/adam.py#L41